### PR TITLE
FOUR-29250 End Event – External URL with Mustache Support

### PR DESCRIPTION
**Pull request by gproly**
Description:
feat(end event): support Mustache in external URL for element destination

- Resolve Mustache expressions in end event "External URL" when the token reaches the end event, using the same context as scripts/screens (APP_URL, _request, _user, process variables).
- Add getElementDestinationMustacheContext() to build context via DataManager with fallback; normalize to plain array for Mustache.
- Add resolveElementDestinationUrl() to decode HTML entities and render URL template with MustacheExpressionEvaluator. FEEL is not supported.
- Apply resolution only for externalURL type; conditional redirect URLs also go through Mustache when destination is external URL.
- Harden getElementDestinationAttribute(): ensure conditionalRedirectProp and elementDestinationProp are never null (json_decode ?? [], pass ?? []).
- Modeler: allow URL validation when string contains {{ (Mustache); update helper and error copy to document _request.id, _user.id, process vars. 
 
Related tickets:
https://processmaker.atlassian.net/browse/FOUR-29250

---

> [!NOTE]
> **Low Risk**
> UI-only validation and copy changes plus a new pure utility with unit tests; no runtime redirect/execution logic is modified.
> 
> **Overview**
> Adds shared URL validation for External URL destinations that accepts either a standard URL or a Mustache-templated URL (e.g. `{{APP_URL}}`, `{{_request.id}}`, `{{_user.id}}`, and process variables), and updates helper/error copy to document the supported placeholders.
> 
> Both `ElementDestination` and Conditional Redirect Task Destination now use the new `isValidElementDestinationURL` utility (with stricter “required only when selected” behavior) and include unit tests covering Mustache parsing edge cases and URL-skeleton validation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c1374e62edbc09efa8d9e4fc30e7cff1558c4188. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/components/inspectors/ElementDestination.vue [hunk 1]
```diff
@@ -174,12 +177,22 @@ export default {
...
+        return '';
+      }
       if (!this.isValidURL(url)) {
-        return this.$t('Must be a valid URL');
+        return this.$t('Must be a valid URL or Mustache expressions ({{APP_URL}}, {{_request.id}}, {{_user.id}}, process variables).');
       }
       return '';
     },
     isValidURL(string) {
+      if (!string || typeof string !== 'string') {
```
Comment (rodriquelca) [inline src/components/inspectors/ElementDestination.vue]: null, undefined, 123, {}, []  will be  valid  for a isValidURL method

File: src/components/inspectors/ElementDestination.vue [hunk 2]
```diff
@@ -174,12 +177,22 @@ export default {
...
+        return true;
+      }
+      // Allow Mustache: same context as backend (APP_URL, _request, _user, process variables)
+      if (string.includes('{{')) {
```
Comment (rodriquelca) [inline src/components/inspectors/ElementDestination.vue]: "{{"
"{{invalid"
"hello {{"

will be evaluated as a valid value, try a regEx  patern 
example

const mustachePattern = /{{\s*[^}]+\s*}}/;

**Review by rodriquelca — APPROVED** [review]
